### ISSUE TITLE
possible sndio error, can't get sound working and errors.

### ISSUE THREAD

**Issue by FrostKnight** [libretro/RetroArch#16224]
# First and foremost consider this:
- Only RetroArch bugs should be filed here. Not core bugs or game bugs
- This is not a forum or a help section, this is strictly developer oriented

This is neither a game bug or a core bug from what I see, its something else.
Seems sndio doesn't work currently on this. I don't know if tiny-alsa alsa-thread or alsa are needed to allow that compatibility, but I tried all 3, none worked.

## Description
I loaded a game via citra and the sound isn't being heard and also I had an error like this:

./RetroArch-Linux-x86_64.AppImage
Setting $HOME to /home/someuser/RetroArch_Qt/RetroArch-Linux-x86_64/RetroArch-Linux-x86_64.AppImage.home
process 27374: D-Bus library appears to be incorrectly set up; failed to read machine uuid: Failed to open "/etc/machine-id": No such file or directory
See the manual page for dbus-uuidgen to correct this issue.
ALSA lib seq_hw.c:466:(snd_seq_hw_open) open /dev/snd/seq failed: No such file or directory
process 27374: D-Bus library appears to be incorrectly set up; failed to read machine uuid: Failed to open "/etc/machine-id": No such file or directory
See the manual page for dbus-uuidgen to correct this issue.
error: XDG_RUNTIME_DIR not set in the environment.
ALSA lib seq_hw.c:466:(snd_seq_hw_open) open /dev/snd/seq failed: No such file or directory

### Expected behavior

Sound is supposed to work.

### Actual behavior

Sound isn't working it looks like

I am using sndio though.

### Steps to reproduce the bug

I loaded your appimage with ./Appimage 

like the above

### Bisect Results

The moment I started the appimage.

### Version/Commit
You can find this information under Information/System Information

I was using Latest Stable

### Environment information

Hyperbola
Was using an appimage

**Comment by gouchi** [comment]
Do you have [alsa-sndio](https://www.hyperbola.info/packages/extra/x86_64/alsa-sndio/) package installed ?

Did you try to check if the issue also happens if you compile it ?

```
git clone https://github.com/libretro/RetroArch.git
cd RetroArch
./configure && make -j$(proc)
./retroarch
```
Thank you.

**Comment by FrostKnight** [comment]
> Do you have [alsa-sndio](https://www.hyperbola.info/packages/extra/x86_64/alsa-sndio/) package installed ?
> 
> Did you try to check if the issue also happens if you compile it ?
> 
> ```
> git clone https://github.com/libretro/RetroArch.git
> cd RetroArch
> ./configure && make -j$(proc)
> ./retroarch
> ```
> 
> Thank you.

Yes I have alsa-sndio, but no I didn't know how to even compile it, because I saw no instructions on how to. But I will try now.

**Comment by gouchi** [comment]
> but no I didn't know how to even compile it, because I saw no instructions on how to. But I will try now.

We have a documentation online using https://docs.libretro.com

And for compilation we have [some information](https://docs.libretro.com/development/retroarch/compilation/ubuntu/#retroarch-compilation).

**Comment by FrostKnight** [comment]
> > but no I didn't know how to even compile it, because I saw no instructions on how to. But I will try now.
> 
> We have a documentation online using https://docs.libretro.com
> 
> And for compilation we have [some information](https://docs.libretro.com/development/retroarch/compilation/ubuntu/#retroarch-compilation).

yeah, sndio does work when i  build it from source, thanks! :)

**Comment by gouchi** [comment]
@FrostKnight Please close this issue.
